### DISCUSSION TITLE
About task_acks_late

### DISCUSSION THREAD

**Discussion by BcnJcaicai** [celery/celery#7958] [Q&A]
Hi all,
         After i  read the doc about the [task_acks_late](https://docs.celeryq.dev/en/stable/glossary.html#term-late-acknowledgment),  i do the test but found something not the same as description, my procedure was the following:

First start the celery worker in one shell and it will consume the default queue task
```
celery -A celery_ini worker -l info
```

Celery App configure like this:
```
from celery import Celery
app = Celery('tasks1', broker="redis://:xxxxx@xxxxx:xxxxx/0",
             backend="mongodb://xxxxx:xxxxx@xxxxx:xxxxx/xxxxx?authSource=xxxxx&maxPoolSize=xxxxx&replicaSet=xxxxx",
             include=["tasks"])
app.conf.update(
    worker_concurrency=2,
    worker_prefetch_multiplier=2,
    task_acks_late=True
)
```

Add task code like this:
```
@app.task
  def add(x, y):
      import time
      time.sleep(30)
      print(f"x+y={x+y}")
      return x + y
```

and i call the task in another shell 
```
from tasks import add
from celery import group
all_tasks=group(add.s(1,i) for i in range(6))
all_tasks.delay()
```

Now ,if as the prefork thread was doing the first 2 job we kill the worker use this cmd 
```
for i in `ps -ef | grep celery | grep -v grep | grep -v flower | awk '{print $2}'`;do kill -9 $i;done;
```   
 
i lost the **exec task** and **reserved task** , the broker was only have **2** task,

but the doc say this， 

> late acknowledgment
Task is [acknowledged](https://docs.celeryq.dev/en/stable/glossary.html#term-acknowledged) after execution (both if successful, or if the task is raising an error), which means the task will be redelivered to another worker in the event of the machine losing power, or the worker instance being killed mid-execution.

so was i misunderstanding this ? can anyone help me ? 

my celery version is 
celery --version 
5.1.2 (sun-harmonics)

[a comment by an automated account is not shown]

**Comment by mapa17** [comment]
Hi @BcnJcaicai I am not sure what your questions is exactly, maybe you can reformulate it.
But based on your description you create 6 sleep tasks, and up to 4 of them will be run at once based on your worker configuration.
If you will workers without terminating the scheduler they after some timeout should be added to your work queue again.
